### ISSUE TITLE
Composum edits of images/files are not visible 

### ISSUE THREAD

**Issue by ieugen** [ist-dresden/composum#84]
I'm using the latest sling-launchpad  `org.apache.sling.launchpad-9-20170531.170627-12089.jar`  with composum browser. 
I use the browser to create a `nt:file` image `content/hello.png`. I can preview the image. 
After that I upload a new image by double clicking on the jcr:data and using the upload dialog there. 

Using the preview button on `jcr:content` node, I can see the new image. Using preview on the `hello.png` file shows the old version. 

Also, viewing the image on the public link displayes the old image as well [1]

One workaround is to manually change the `jcr:lastModified` property of the 'jcr:content' node. 
That way, the image is updated. 

The same thing happens with file updates as well. Editing a file via the composum web interface does not change the contents in the public link, unless you update the `jcr:lastModified` proeprty.

p.s. I will post this to the sling dev mailing list because I'm not sure where to post. I haven't seen where is the best way. 

[1] http://localhost:8080/content/hello.png

**Comment by ist-rw** [comment]
1. Yes, the image viewer doesn't display the latest data of an image after upload into the 'jcr:data' property. This is accepted as a bag if no modifiers are specified in the display view. I've fixed this but if modifiers are present (resolver mapping in 'on', Sling selectors or URL parameters are specified) the URL is built like the URL in a Sling based application and the behaviour is the same as before - the new image is visible only if the 'ifModifiedSince' condition of the request is true.
2. The property editor itself should not change another property if a property like 'jcr:data' is changed. The browser doesn't now dependencies between properties - he is working on the raw attribute level. But the file views (the image view is one of them) have now an upload button in the views toolbar to change the 'jcr:data' property of a file or its content resource and the corresponding upload dialog has now an additional checkbox - adjust 'jcr:modified' properties - to synchronise the upload of new file data with a 'touch'. This is available in the upcoming release 1.8 - i hope it helps.

**Comment by ieugen** [comment]
Hello @ist-rw, 

Thanks for the information. 
This behavior also affects the usability of Composum. Composum is meant to be an Explorer for Sling so people can view *and* edit content via a web interface. 

However, if the content is not visible after the edit, not only in Composum, but also on the real website, than that is, in my opinion a bug. I imagine Composum  *should* change the *jcr:modified* property if a property of the node is changed. 
From a user point of view, it kind of makes sense. I have changed a property of a node, then I have changed the node so *jcr:modified* should be changed. If this is what Composum does/will do, I believe it's good. 

Regarding the check for file uploads, I recommend having the check enabled by default. As a user, I expect to be able to view the files on via Composum and the web application, after I upload them. 

p.s. Any ETA on 1.8? Any way I can help? I haven't seen a public issue tracker other than this one. and it's not a lot of information regarding the roadmap tasks. 

Thanks,
